src/validr/_validator.pyx

```diff
@@ -129,14 +130,21 @@ def list_validator(compiler, items=None, int minlen=0, int maxlen=1024,
         except TypeError:
             raise Invalid('not list')
         result = []
+        keys = set()
         cdef int i = -1
         for i, x in value:
             if i >= maxlen:
                 raise Invalid('list length must <= %d' % maxlen)
             with mark_index(i):
                 v = inner(x) if inner is not None else copy.deepcopy(x)
-                if unique and v in result:
-                    raise Invalid('not unique')
+                if unique:
+                    if unique_key_helper:
+                        key = compiler.helpers.get(unique_key_helper)(v)
```

Comment:
`compiler.helpers.get(unique_key_helper)` should run in compile time, and raise SchemaError when unique_key_helper not exists. (raise error earlier, find bug easier)

Reply:
Okay, good point, but i've not digged deep into your code, so at compile time is - when?

Reply:
between `def list_validator` and `def validate` --- the code run when we call `compiler.compile`